Typing test error in maintenance/1.21.x

@BvB93 I'm seeing the following test failure after this:
```
_____________________________________________________ test_reveal[arrayterator.py] _____________________________________________________

path = '/home/charris/Workspace/numpy.git/build/testenv/lib64/python3.9/site-packages/numpy/typing/tests/data/reveal/arrayterator.py'
reveal = "'typing.Generator[numpy.signedinteger[numpy.typing._64Bit], None, None]"
expected_reveal = '13: note: Revealed type is "typing.Generator[numpy.signedinteger[numpy.typing._64Bit], None, None]"', lineno = 13

    def _test_reveal(path: str, reveal: str, expected_reveal: str, lineno: int) -> None:
        if reveal not in expected_reveal:
>           raise AssertionError(_REVEAL_MSG.format(lineno, expected_reveal, reveal))
E           AssertionError: Reveal mismatch at line 13
E           
E           Expected reveal: '13: note: Revealed type is "typing.Generator[numpy.signedinteger[numpy.typing._64Bit], None, None]"'
E           Observed reveal: "'typing.Generator[numpy.signedinteger[numpy.typing._64Bit], None, None]"

expected_reveal = '13: note: Revealed type is "typing.Generator[numpy.signedinteger[numpy.typing._64Bit], None, None]"'
lineno     = 13
path       = '/home/charris/Workspace/numpy.git/build/testenv/lib64/python3.9/site-packages/numpy/typing/tests/data/reveal/arrayterator.py'
reveal     = "'typing.Generator[numpy.signedinteger[numpy.typing._64Bit], None, None]"
```
This is with python 3.9.7. This also shows up in 1.21.3 (previous release) and seems new, probably something in my environment has changed.

--- a/numpy/typing/tests/data/reveal/arrayterator.py
+++ b/numpy/typing/tests/data/reveal/arrayterator.py
@@ -10,7 +10,7 @@ reveal_type(ar_iter.start)  # E: builtins.list[builtins.int]
 reveal_type(ar_iter.stop)  # E: builtins.list[builtins.int]
 reveal_type(ar_iter.step)  # E: builtins.list[builtins.int]
 reveal_type(ar_iter.shape)  # E: builtins.tuple[builtins.int]
-reveal_type(ar_iter.flat)  # E: 'typing.Generator[{int64}, None, None]
+reveal_type(ar_iter.flat)  # E: typing.Generator[{int64}, None, None]
 
 reveal_type(ar_iter.__array__())  # E: numpy.ndarray[Any, numpy.dtype[{int64}]]
 

